dom0-updates/qubes-dom0-update

```diff
@@ -208,22 +215,43 @@ echo "Using $UPDATEVM as UpdateVM to download updates for Dom0; this may take so
 qvm-run --nogui -q -u root "$UPDATEVM" 'mkdir -m 775 -p /var/lib/qubes/dom0-updates/' || exit 1
 qvm-run --nogui -q -u root "$UPDATEVM" 'chown user:user /var/lib/qubes/dom0-updates/' || exit 1
 qvm-run --nogui -q "$UPDATEVM" 'rm -rf /var/lib/qubes/dom0-updates/etc' || exit 1
-tar c /var/lib/rpm /etc/yum.repos.d /etc/yum.conf /etc/dnf/dnf.conf 2>/dev/null | \
-   qvm-run --nogui -q --pass-io "$UPDATEVM" 'LC_MESSAGES=C tar x -C /var/lib/qubes/dom0-updates 2>&1 | grep -v -E "s in the future"'
-
-CMD="/usr/lib/qubes/qubes-download-dom0-updates.sh --doit --nogui $(escape_args "${ALL_OPTS[@]}")"
-# Use 'script' to fake a terminal, so that we are shown sync and download
-# progress indicators. However, do it only if we are running in terminal
-# ourselves.
-if [ -t 1 ] && [ -t 2 ]; then
-    CMD="script --quiet --return --command '$CMD' /dev/null"
+tar c /var/lib/rpm /etc/yum.repos.d /etc/yum.conf /etc/dnf/dnf.conf 2>/dev/null |
+   qvm-run --nogui -q --pass-io "$UPDATEVM" 'LC_MESSAGES=C tar x -C /var/lib/qubes/dom0-updates' >/dev/null 2>&1 || {
+   status=$?
+   echo "Sending repository information to UpdateVM failed: code $status">&2
+   exit "$status"
+}
+
+CMD="/usr/lib/qubes/qubes-download-dom0-updates.sh --doit --nogui"
+
+# We avoid using bash’s own facilities for this, as they produce $'\n'-style
+# strings in certain cases.  These are not portable, whereas the string produced
+# by the following is.
+for i in "${ALL_OPTS[@]}"; do CMD+=" '${i//\'/\'\\\'\'}'"; done
+
+# If we are running in a terminal, run the command in its own terminal in the UpdateVM.
+if [[ "$CONSOLE" = 1 ]]; then
+    QVMRUN_OPTS=(--nogui)
+    if [[ -t 1 ]] && [[ -t 2 ]]; then
+        # Use ‘script’ to emulate a TTY, so that we get status bars and other
+        # progress output.  Since stdout and stderr are both terminals, qvm-run
+        # will automatically sanitize them, but we explicitly tell it to anyway
+        # as a precaution.
+        CMD="exec script --quiet --return --command '${CMD//\'/\'\\\'\'}'"
+        QVMRUN_OPTS+=(--pass-io --filter-escape-chars --color-stdout=31 --color-stderr=31)
```

Comment:
No need to force `--color-*` options, they will be set based on `--filter-escape-chars` and being on terminal or not (i.e. do not output garbage if output is not a terminal that understands coloring sequences).